public/index.js

```diff
@@ -0,0 +1,133 @@
+const URL = "https://ada-backtrek-api.herokuapp.com/trips"
+
+const reportStatus = (message) => {
+  $('#status-message').html(message);
+};
+
+const reportError = (message, errors) => {
+  let content = `<p>${message}</p><ul>`;
+  for (const field in errors) {
+    for (const problem of errors[field]) {
+      content += `<li>${field}: ${problem}</li>`;
+    }
+  }
+  content += "</ul>";
+  reportStatus(content);
+};
+
+const loadTrips = () => {
+  reportStatus('Loading trips...');
+
+  const tripList = $('#trip-list');
+  tripList.empty();
+
+  axios.get(URL)
+  .then((response) => {
+    reportStatus(`Successfully loaded ${response.data.length} trips`);
+    $('.box-header').remove();
+    tripList.before(`<h2 class="box-header">All Trips</h2>`);
+    response.data.forEach((trip) => {
+      tripList.append(`<li class="trip" id="${trip.id}" tabindex="0">${trip.name}</li>`);
+    });
+  })
+  .catch((error) => {
+    reportError(`There was a problem while loading trips: ${error.message}`);
+    console.log(error);
+  });
+};
+
+const loadTrip = function loadTrip() {
+  reportStatus('Loading trips...');
+  const showTrip = $('#show-trip');
+  showTrip.empty();
+  const reserveTrip = $('#reserve-trip');
+  reserveTrip.empty();
+
+  let trip_id = this.id;
+
+  axios.get(URL + `/${trip_id}`)
+  .then((response) => {
+    reportStatus(`Successfully loaded trip with id: ${response.data.id}`);
+    $('#trip-details').remove();
+    showTrip.before(`<h2 class="box-header" id="trip-details">Trip Details</h2>`);
+
+    let trip = response.data
+    let cost = parseInt(trip.cost).toLocaleString('en-US', {style: 'currency', currency: 'USD' });
+    showTrip.html(`
+      <li><strong>Name:</strong> ${trip.name}</li>
+      <li><strong>Continent:</strong> ${trip.continent}</li>
+      <li><strong>Category:</strong> ${trip.category}</li>
+      <li><strong>Weeks:</strong> ${trip.weeks}</li>
+      <li><strong>Cost:</strong> ${cost}</li>
+      <li class='about'><strong>About:</strong> ${trip.about}</li>
+      `
+    );
+
+    reserveTrip.html(`
+      <h2 class="box-header">Reserve Trip: </h2>
+      <form id="reservation-form">
+      <div>
+      <label for="name">Name</label>
+      <input type="text" name="name" />
+      </div>
+
+      <div>
+      <label for="email">Email</label>
+      <input type="text" name="email" />
+      </div>
+
+      <div id="trip-name">
+      <strong>Trip Name:</strong> ${trip.name}
+      </div>
+
+      <input type="submit" name="reserve-trip" value="make reservation" />
+      </form>
+      `
+    );
+
+    const createReservation = (event) => {
```

Comment:
By having this function directly inside the function that handles loading a single trip, this greatly reduces the readability, especially in trying to find the `catch`. Consider moving this out and accessing the trip's ID in a different way - preferrably via the DOM.